akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -789,6 +789,9 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
         leaderActionCounter = 0
         leaderActionsOnConvergence()
       } else {
+        if (cluster.settings.AllowWeaklyUpMembers)
```

Comment:
No, it will be done on both sides of a network partition. The leader is the lowest address of the reachable nodes, i.e. there is always a leader, and there may be several. It is convergence that controls that one leader performs certain tasks, but this task is performed by all leaders.